test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/TestModels/InheritanceModels.cs

```diff
@@ -54,6 +54,42 @@ public override int WheelCount
 
         [NotMapped]
         public IEnumerable<MotorcycleManufacturer> Manufacturers { get; set; }
+
+        public Engine MyEngine { get; set; }
+
+        public V4 MyV4Engine { get; set; }
+    }
+
+    public class Engine
+    {
+        public int Hp { get; set; }
+    }
```

Comment:
can we add another complex property to the base "engine" type, that has derived types, so that we can verify that the recursion works?